pyomo/contrib/pynumero/interfaces/tests/test_nlp.py

```diff
@@ -10,660 +10,609 @@
+    ret = anlp.evaluate_jacobian_eq(out=jac_eq)
+    self.assertTrue(ret is jac_eq)
+    dense_jac_eq = jac_eq.todense()
+    self.assertTrue(np.array_equal(dense_jac_eq, expected_jac_eq))
+    # change the value of the primals
+    # ToDo: not a great test since this problem is linear
+    anlp.set_primals(2.0*np.ones(9))
+    anlp.evaluate_jacobian_eq(out=jac_eq)
+    dense_jac_eq = jac_eq.todense()
+    self.assertTrue(np.array_equal(dense_jac_eq, expected_jac_eq))
+
+    # jacobian of inequality constraints
+    jac_ineq = anlp.evaluate_jacobian_ineq()
+    dense_jac_ineq = jac_ineq.todense()
+    expected_jac_ineq = [ [(i)*(j) for j in range(1,10)] for i in [1, 3, 4, 5, 7, 8, 9] ]
+    expected_jac_ineq = np.asarray(expected_jac_ineq, dtype=np.float64)
+    self.assertTrue(np.array_equal(dense_jac_ineq, expected_jac_ineq))
+    # test inplace
+    jac_ineq.data = 0*jac_ineq.data
+    ret = anlp.evaluate_jacobian_ineq(out=jac_ineq)
+    self.assertTrue(ret is jac_ineq)
+    dense_jac_ineq = jac_ineq.todense()
+    self.assertTrue(np.array_equal(dense_jac_ineq, expected_jac_ineq))
+    # change the value of the primals
+    # ToDo: not a great test since this problem is linear
+    anlp.set_primals(2.0*np.ones(9))
+    anlp.evaluate_jacobian_ineq(out=jac_ineq)
+    dense_jac_ineq = jac_ineq.todense()
+    self.assertTrue(np.array_equal(dense_jac_ineq, expected_jac_ineq))
+
+    # hessian
+    hess = anlp.evaluate_hessian_lag()
+    dense_hess = hess.todense()
+    expected_hess = [ [2.0*i*j for j in range(1, 10)] for i in range(1,10) ]
+    expected_hess = np.asarray(expected_hess, dtype=np.float64)
+    self.assertTrue(np.array_equal(dense_hess, expected_hess))
+    # test inplace
+    hess.data = np.zeros(len(hess.data))
+    ret = anlp.evaluate_hessian_lag(out=hess)
+    self.assertTrue(ret is hess)
+    dense_hess = hess.todense()
+    self.assertTrue(np.array_equal(dense_hess, expected_hess))
+    # change the value of the primals
+    anlp.set_primals(2.0*np.ones(9))
+    anlp.evaluate_hessian_lag(out=hess)
+    dense_hess = hess.todense()
+    self.assertTrue(np.array_equal(dense_hess, expected_hess))
+    # change the value of the obj factor
+    anlp.set_obj_factor(2.0)
+    hess = anlp.evaluate_hessian_lag()
+    dense_hess = hess.todense()
+    expected_hess = [ [4.0*i*j for j in range(1, 10)] for i in range(1,10) ]
+    expected_hess = np.asarray(expected_hess, dtype=np.float64)
+    self.assertTrue(np.array_equal(dense_hess, expected_hess))
 
-def create_rosenbrock_model(n_vars):
-
-    model = pe.ConcreteModel()
-    model.nx = n_vars
-    model.lb = -10
-    model.ub = 20
-    model.init = 2.0
-    model.index_vars = range(n_vars)
-    model.x = pe.Var(model.index_vars, initialize=model.init, bounds=(model.lb, model.ub))
-
-    def rule_constraint(m, i):
-        return m.x[i]**3-1.0 == 0
-    model.c = pe.Constraint(model.index_vars, rule=rule_constraint)
+@unittest.skipIf(os.name in ['nt', 'dos'], "Do not test on windows")
```

Comment:
I accidentally deleted a comment while finishing up my review. I had asked what part of PyNumero is not supported on Windows and @carldlaird replied with:

 Currently, I believe it is the pynumero_libraries that provide interfaces to ASL, etc. @eslickj may have a fix for this (he recently got PyNumero running on Windows). We would need to get these fixes into the conda recipe.

@santiagoropb, @eslickj: Is this interpretation correct?

Reply:
Yeah, I can run Pynumero on windows no problem.  I used MinGW to compile the ASL part, but I have no idea about conda recipes.  I did make a PR to make a small fix to some types, and I think it got merged in the mainline Pyomo.  That was the only difficulty I had on Windows.

I'd be happy to help however I can.  We definitely will need to an easy solution to use it on Windows.

Reply:
@eslickj thanks for the clarification. I don't think this should hold up this PR so I'll open a new issue to track PyNumero support on Windows.

Reply:
@blnicho @eslickj The only c++ library that is needed for this PR to work on windows is the asl  one. I have a CMAKE project that builds on windows, linux and mac. I believe the library built in conda-forge works with MinGW without any additional changes. However, for complete support of Windows my latest version of the CMAKE project should be updated in the recipe. I have tested that CMAKE project with visual studio 2017 on a windows machine and it worked fine. Should I send you the cmake project via email?

Reply:
If I can get it with conda, I think I'm good.  I don't need to build it myself, unless there is a problem that needs sorted out.